pkg/apis/crd/v1alpha2/types.go

```diff
@@ -543,3 +543,94 @@ type TrafficControlList struct {
 
 	Items []TrafficControl `json:"items"`
 }
+
+type BGPPolicyStatus string
+
+const (
+	BGPPolicyActive  = "Active"
+	BGPPolicyStandby = "Standby"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BGPPolicy defines BGP configuration applied to Nodes.
+type BGPPolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec BGPPolicySpec `json:"spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type BGPPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []BGPPolicy `json:"items"`
+}
+
+// BGPPolicySpec defines the specification for a BGP policy.
+type BGPPolicySpec struct {
+	// NodeSelector is utilized to select Nodes to which the policy is applied.
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector"`
+
+	// LocalASN is the AS number used by BGP protocol.
+	LocalASN int16 `json:"localASN"`
+
+	// BindAddresses is a list of IP addresses binding to BGP protocol.
+	BindAddresses []string `json:"bindAddresses,omitempty"`
+
+	// ListenPort is the port on which BGP protocol listen.
+	ListenPort *int16 `json:"listenPort,omitempty"`
+
+	// RouterID is the routerID used by BGP protocol in the format of an IPv4 address. This field is required on an IPv6
+	// single stack cluster.
+	RouterID *string `json:"routerID,omitempty"`
+
+	// Advertisements selects routes to be advertised to external BGP peers.
+	Advertisements Advertisements `json:"advertisements"`
+
+	// BGPPeers are the list of external BGP peers.
+	BGPPeers []BGPPeer `json:"bgpPeers"`
+}
+
+type Advertisements struct {
```

Comment:
just to double check: if all boolean values are omitted (default), then we won't advertise any IP?

Reply:
Yes, I think this is a legal case since establishing with external BGP peers but not advertising any IP is allowed.